src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -0,0 +1,171 @@
+                memberSymbol.GetGlyph(),
+                baseSymbolItems.Concat(derivedTypeItems));
+        }
+
+        private static async Task<InheritanceTargetItem> CreateInheritanceItemAsync(
+            Solution solution,
+            ISymbol targetSymbol,
+            InheritanceRelationship inheritanceRelationshipWithOriginalMember,
+            CancellationToken cancellationToken)
+        {
+            var definition = await targetSymbol.ToClassifiedDefinitionItemAsync(
+                solution,
+                isPrimary: true,
+                includeHiddenLocations: false,
+                FindReferencesSearchOptions.Default,
+                cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            return new InheritanceTargetItem(
+                inheritanceRelationshipWithOriginalMember,
+                definition);
+        }
+
+        private static async Task<InheritanceMemberItem> CreateInheritanceMemberInfoForMemberAsync(
+            Solution solution,
+            ISymbol memberSymbol,
+            int lineNumber,
+            ImmutableArray<ISymbol> implementingMembers,
+            ImmutableArray<ISymbol> implementedMembers,
+            ImmutableArray<ISymbol> overridenMembers,
+            ImmutableArray<ISymbol> overridingMembers,
+            CancellationToken cancellationToken)
+        {
+            var implementingMemberItems = await implementingMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implementing, cancellationToken)).ConfigureAwait(false);
+            var implementedMemberItems = await implementedMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implemented, cancellationToken)).ConfigureAwait(false);
+            var overridenMemberItems = await overridenMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Overriden, cancellationToken)).ConfigureAwait(false);
+            var overridingMemberItems = await overridingMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Overriding, cancellationToken)).ConfigureAwait(false);
+
+            return new InheritanceMemberItem(
+                lineNumber,
+                memberSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                memberSymbol.GetGlyph(),
+                implementingMemberItems.Concat(implementedMemberItems)
+                    .Concat(overridenMemberItems)
+                    .Concat(overridingMemberItems));
+        }
+
+        /// <summary>
+        /// For the <param name="memberSymbol"/>, get all the symbols implementing it.
+        /// </summary>
+        private static async Task<ImmutableArray<ISymbol>> GetImplementedSymbolsAsync(
```

Comment:
```suggestion
        private static async Task<ImmutableArray<ISymbol>> GetImplementingSymbolsAsync(
```

Reply:
: ) It should be implemented, since the 'implemented' is the down arrow. I'll update the comments here

Reply:
`I↓` indicates that the current symbol (`memberSymbol`) is implemented by one or more other symbols. Those other symbols are the implementing symbols.

Reply:
I'll try to send a small PR tomorrow to make all the cases consistent throughout this PR. Will be easier than itemizing them all in review.

Reply:
I think I found the gap, KnowMoniker.Implemented is I↓ and I basically follow its definition. And I think you treat is in the opsite way.